src/modules/75-cf/components/FlagActivation/FlagArchiving/CannotArchiveWarning.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { FC, useMemo } from 'react'
+import { Layout, TableV2, Text } from '@harness/uicore'
+import { FontVariation } from '@harness/design-system'
+import { Link, useParams } from 'react-router-dom'
+import type { CellProps, Column } from 'react-table'
+import useActiveEnvironment from '@cf/hooks/useActiveEnvironment'
+import routes from '@common/RouteDefinitions'
+import { useStrings } from 'framework/strings'
+import type { Feature, Prerequisite } from 'services/cf'
+
+export interface CannotArchiveWarningProps {
+  environment: string
+  flagData: Feature
+}
+
+const CannotArchiveWarning: FC<CannotArchiveWarningProps> = ({ environment, flagData }) => {
+  const { getString } = useStrings()
+  const { withActiveEnvironment } = useActiveEnvironment()
+  const { projectIdentifier, orgIdentifier, accountId } = useParams<Record<string, string>>()
+
+  const columns: Column<Prerequisite>[] = useMemo(
+    () => [
+      {
+        Header: '',
+        width: '70%',
+        id: 'flagDetails',
+        Cell: ({ row }: CellProps<Prerequisite>) => {
+          return (
+            <Link
+              to={withActiveEnvironment(
+                routes.toCFFeatureFlagsDetail({
+                  orgIdentifier: orgIdentifier as string,
+                  projectIdentifier: projectIdentifier as string,
+                  featureFlagIdentifier: row.original.feature,
+                  accountId
+                }),
+                environment
+              )}
```

Comment:
As above, since any page you're on and invoking this will already have an active environment, you might as well just use `withActiveEnvironment` without the optional env override.
```suggestion
              to={withActiveEnvironment(
                routes.toCFFeatureFlagsDetail({
                  orgIdentifier: orgIdentifier as string,
                  projectIdentifier: projectIdentifier as string,
                  featureFlagIdentifier: row.original.feature,
                  accountId
                })
              )}
```